compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BStructureType.java

```diff
@@ -49,4 +54,22 @@ public BStructureType(int tag, BTypeSymbol tSymbol, int flags) {
     public void accept(TypeVisitor visitor) {
         visitor.visit(this);
     }
+
+    protected boolean shouldPrintShape(Name name) {
+        if (name == null) {
+            return false;
+        }
+
+        String value = name.value;
+
+        if (value.isEmpty() || value.startsWith(DOLLAR)) {
+            return true;
+        }
+
+        if (!(Symbols.isFlagOn(this.tsymbol.flags, Flags.READONLY))) {
+            return false;
+        }
+
+        return value.startsWith("($");
```

Comment:
Just curious what this check does?

Reply:
For immutable record/object types the name is something like `(T & readonly)`. So in the anonymous case it will be something like `($anonType$n & readonly)`.

We should clean this up in general though, to look at a flag maybe, instead of the name. https://github.com/ballerina-platform/ballerina-lang/issues/24471 tracks that.